code/modules/events/tear.dm

```diff
@@ -1,47 +1,93 @@
+// Dimensional Tear - A rift appears randomly on the station, does the following:
+// Flickers nearby machines as an early warning.
+// After a few seconds, it breaks nearby computers and mirrors.
+// A portal appears, then it spawns a few hostile mobs, and a leader one.
+// Then the portal deletes itself.
+
+// Event setup
 /datum/event/tear
-	startWhen = 3
-	announceWhen = 20
-	endWhen = 50
+	name = "dimensional tear"
+	announceWhen = 6
+	endWhen = 10
 	var/obj/effect/tear/TE
 
-/datum/event/tear/announce()
-	GLOB.event_announcement.Announce("A tear in the fabric of space and time has opened. Expected location: [impact_area.name].", "Anomaly Alert", 'sound/AI/anomaly.ogg')
+/datum/event/tear/setup()
+	impact_area = findEventArea()
 
 /datum/event/tear/start()
 	var/turf/T = pick(get_area_turfs(impact_area))
 	if(T)
-		TE = new /obj/effect/tear(T.loc)
+		// Give ghosts some time to jump there before it begins.
+		var/image/alert_overlay = image('icons/mob/animal.dmi', "hellhound")
+		notify_ghosts("\A [src] is about to open in [get_area(T)].", title = "Dimensional Rift", source = T, alert_overlay = alert_overlay, action = NOTIFY_FOLLOW)
+		addtimer(CALLBACK(src, .proc/spawn_tear, T), 4 SECONDS)
 
-/datum/event/tear/setup()
-	impact_area = findEventArea()
+		// Energy overload; we mess with machines as an early warning and for extra spookiness.
+		for(var/obj/machinery/M in range(8, T))
+			INVOKE_ASYNC(M, /obj/machinery/get_spooked)
```

Comment:
```suggestion
			INVOKE_ASYNC(M, .proc/get_spooked)
```
Try this. I remember having an issue with invoke async and the linter a while ago

Reply:
Sadly, this returns an error:
`code\modules\events\tear.dm:27:error: .get_spooked: undefined type path`

Reply:
I had to call the parent proc with `INVOKE_ASYNC(M, /atom/.proc/get_spooked)`, the linter is now satisfied